src/lib.rs

```diff
@@ -82,11 +84,20 @@ pub use poll::{Poll, Registry};
 pub use token::Token;
 pub use waker::Waker;
 
-#[cfg(all(unix, feature = "os-util"))]
-#[cfg_attr(docsrs, doc(cfg(all(unix, feature = "os-util"))))]
+#[cfg(all(unix, any(feature = "os-util", feature = "pipe")))]
+#[cfg_attr(
+    docsrs,
+    doc(cfg(all(unix, any(feature = "os-util", feature = "pipe"))))
+)]
 pub mod unix {
     //! Unix only extensions.
+
+    #[cfg(feature = "os-util")]
+    #[cfg_attr(docsrs, doc(cfg(all(unix, feature = "os-util"))))]
     pub use crate::sys::SourceFd;
+    #[cfg(all(feature = "pipe", feature = "os-poll"))]
+    #[cfg_attr(docsrs, doc(cfg(all(unix, feature = "pipe", feature = "os-poll"))))]
+    pub use crate::sys::{pipe, Receiver, Sender};
```

Comment:
Having pipe handles named `Receiver` and `Sender` in `unix` is a bit unclear. I would either have it be `unix::pipe::Receiver` or `unix::PipeReceiver` (same w/ sender).

Reply:
I've created a new module `unix::pipe`, with `Sender`, `Receiver` and `new`. So it can be created with `pipe::new`, and receiver accessed with `pipe::Sender`.